FIX Treat gradient boosting categoricals outside the bounds as unknown during predict (#24283)
scikit-learn/scikit-learn: doc/whats_new/v1.2.rst

  NaN in feature importance when fitted with very small sample weight.
  :pr:`20415` by :user:`Zhehao Liu <MaxwellLZH>`.

> - |Fix| :class:`ensemble.HistGradientBoostingClassifier` and
>   :class:`ensemble.HistGradientBoostingRegressor` does not error when predicting
>   on an categories encoded as a negative value and considers it unknown.
>   :pr:`24283` by `Thomas Fan`_.

Reviewer: ```suggestion
- |Fix| :class:`ensemble.HistGradientBoostingClassifier` and
  :class:`ensemble.HistGradientBoostingRegressor` no longer error when predicting
  on categories encoded as negative values and instead consider them unknown.
  :pr:`24283` by `Thomas Fan`_.
```

  NaN in feature importance when fitted with very small sample weight.
  :pr:`20415` by :user:`Zhehao Liu <MaxwellLZH>`.

- |Fix| :class:`ensemble.HistGradientBoostingClassifier` and
  :class:`ensemble.HistGradientBoostingRegressor` no longer error when predicting

>   on categories encoded as negative values and instead consider them unknown.

Reviewer 2: ```suggestion
  on categories encoded as negative values and instead consider them a member of the "missing category".
```
